Model/src/main/java/org/apidb/apicommon/model/report/bed/feature/GenomicSequenceFeatureProvider.java

```diff
@@ -63,12 +76,49 @@ public List<List<String>> getRecordAsBedFields(RecordInstance record) throws Wdk
       defline.appendPosition(chrom, segmentStart, segmentEnd, _strand);
     }
     if(_requestedDeflineFields.contains("ui_choice")){
-      defline.appendGenomicFeatureUiChoice("Whole sequence", _strand);
+      defline.appendGenomicSequenceRangeUiChoice(
+        "Sequence Region",
+        getPositionDesc(_startOffset, _startAnchor),
+        getPositionDesc(_endOffset, _endAnchor),
+        _strand
+      );
     }
     if(_requestedDeflineFields.contains("segment_length")){
       defline.appendSegmentLength(segmentStart, segmentEnd);
     }
     return List.of(BedLine.bed6(chrom, segmentStart, segmentEnd, defline, _strand));
   }
 
+  private static Integer getPosition(Integer featureLength, int offset, Anchor anchor) throws WdkModelException {
+    switch(anchor){
+      case DownstreamFromStart: return 1 + offset;
+      case UpstreamFromEnd: return featureLength - offset;
+      default: throw new WdkModelException("Unsupported anchor type: " + anchor);
+    }
+  }
+
```

Comment:
I feel like the next method can be much more succinct.  This does the same thing, right?
```
  private static String getPositionDesc(int offset, Anchor anchor) throws WdkModelException {
    switch(anchor){
      case DownstreamFromStart:
        return "Start" + (offset > 0 ? "+" + offset : "");
      case UpstreamFromEnd:
        return "End" + (offset > 0 ? "-" + offset : "");
      default:
        throw new WdkModelException("Unsupported anchor type: " + anchor);
    }
  }
```

Reply:
Thanks, your version is neater! Updated the code.